torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -33,7 +33,7 @@ auto AccumulateGrad::apply(variable_list&& grads) -> variable_list {
   if (!variable.requires_grad())
     return {};
 
-  at::Tensor& grad = variable.grad();
+  at::Tensor& grad = variable.mutable_grad();
```

Comment:
This looks wrong, we should call `mutable_grad` after we have taken out the lock